crates/nu-command/src/conversions/split_cell_path.rs

```diff
@@ -0,0 +1,168 @@
+use nu_engine::command_prelude::*;
+use nu_protocol::{ast::PathMember, IntoValue};
+
+#[derive(Clone)]
+pub struct SubCommand;
+
+impl Command for SubCommand {
+    fn name(&self) -> &str {
+        "split cell-path"
+    }
+
+    fn signature(&self) -> Signature {
+        Signature::build(self.name())
+            .input_output_types(vec![
+                (Type::CellPath, Type::List(Box::new(Type::Any))),
+                (
+                    Type::CellPath,
+                    Type::List(Box::new(Type::Record(
+                        [("value".into(), Type::Any), ("optional".into(), Type::Bool)].into(),
+                    ))),
+                ),
+            ])
+            .category(Category::Conversions)
+            .allow_variants_without_examples(true)
+    }
+
+    fn description(&self) -> &str {
+        "Split a cell-path into its components."
+    }
+
+    fn search_terms(&self) -> Vec<&str> {
+        vec!["convert"]
+    }
+
+    fn run(
+        &self,
+        _engine_state: &EngineState,
+        _stack: &mut Stack,
+        call: &Call,
+        input: PipelineData,
+    ) -> Result<PipelineData, ShellError> {
+        let head = call.head;
+
+        match input {
+            PipelineData::Value(Value::CellPath { val, .. }, _) => {
+                Ok(split_cell_path(val, head)?.into_pipeline_data())
+            }
+            PipelineData::Value(other, _) => Err(ShellError::OnlySupportsThisInputType {
+                exp_input_type: "cell-path".into(),
+                wrong_type: other.get_type().to_string(),
+                dst_span: head,
+                src_span: other.span(),
+            }),
+            PipelineData::ListStream(stream, ..) => Err(ShellError::OnlySupportsThisInputType {
+                exp_input_type: "cell-path".into(),
+                wrong_type: "list".into(),
```

Comment:
Wouldn't we need for this to be `stream.type_().describe().into()` so it returns `list<stream>` versus just `list`. Not sure if `get_type()` also works here or not? Also, it seems like you can combine a few of these, at least the streams into one match arm, right?

Reply:
I originally tried
```
PipelineData::ListStream(stream, ..) | PipelineData::ByteStream(stream, ..) => { }
```
but the types of `stream`s don't match.

ListStream (nu_protocol::pipeline::list_stream::ListStream) doesn't have a method similar to `describe` or `get_type` as far as I can tell. `"list".into()` _does_ feel not quite correct but I don't know what else to go with.

Reply:
Managed to simplify this by switching to `PipelineMismatch` as the error variant.